1.5/Source/PawnStorages/PawnStoragesModule-DefensivePositions/CompPawnStorage_Patch.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using DefensivePositions;
+using HarmonyLib;
+using RimWorld;
+using UnityEngine;
+using Verse;
+
+namespace PawnStorages.DefensivePositions;
+
+[HarmonyPatch(typeof(CompPawnStorage))]
+public static class CompPawnStorage_Patch
+{
+    [HarmonyPatch(nameof(CompPawnStorage.CompGetGizmosExtra))]
+    [HarmonyPostfix]
+    public static void CompGetGizmosExtra_Patch(CompPawnStorage __instance, ref IEnumerable<Gizmo> __result)
```

Comment:
If you make this a pass-through postfix you can just yield the extra gizmo

```csharp
public static IEnumerable<Gizmo> CompGetGizmosExtra_Patch(ref IEnumerable<Gizmo> __result, CompPawnStorage __instance)
foreach (Gizmo g in __result) yield return g;
...
yield return action;
```

Only a minor style point really but it does prevent an extra allocation and means that bit at the bottom goes away. It's technically marginally safer too because if our code explodes the other gizmos still returned just fine.